### ISSUE TITLE
`_get_property_list`-example code does not work

### ISSUE THREAD

**Issue by stephanbogner** [godotengine/godot#75679]
### Godot version

v4.0.stable.official [92bee43ad]

### System information

Linux Ubuntu

### Issue description

**[The example for `_get_property_list` is not working](https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-get-property-list).
-> The conditional property `hammer_type` is never shown**

This might be a "docs"-issue but it seems more that there is an underlying bug. Because if I read the docs correctly, the code itself _should_ work. This might be caused due to one of the following issues: #56614, #64227

**Code:**
```
@tool
extends Node2D

@export var holding_hammer = false:
    set(value):
        holding_hammer = value
        notify_property_list_changed()
var hammer_type = 0

func _get_property_list():
    # By default, `hammer_type` is not visible in the editor.
    var property_usage = PROPERTY_USAGE_NO_EDITOR

    if holding_hammer:
        property_usage = PROPERTY_USAGE_DEFAULT

    var properties = []
    properties.append({
        "name": "hammer_type",
        "type": TYPE_INT,
        "usage": property_usage, # See above assignment.
        "hint": PROPERTY_HINT_ENUM,
        "hint_string": "Wooden,Iron,Golden,Enchanted"
    })

    return properties
```

**Result: `hammer_type` is not shown in editor**
![grafik](https://user-images.githubusercontent.com/7897006/230026565-faff820f-7528-46bf-ab6c-4bbe37094b55.png)

### Steps to reproduce

1. Setup basic project
2. Copy code from [here](https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-get-property-list) 
3. Toggle `holding_hammer`
4. Nothing happens

### Minimal reproduction project

[PropertyListBug.zip](https://github.com/godotengine/godot/files/11156800/PropertyListBug.zip)

**Comment by AThousandShips** [comment]
You need to add a `_get` function handling it as well, see the documentation for that

**Comment by AThousandShips** [comment]
You need to reload the scene, then it works, at least for me

**Comment by stephanbogner** [comment]
> You need to reload the scene, then it works, at least for me

Indeed after restarting Godot, it works now for me too. Thanks a lot!
I tried for 2 hours but didn't think of that :sweat_smile:

**Comment by AThousandShips** [comment]
At the moment `@tool` scripts generally need the editor to restart to work, not sure why

**Comment by jupiterbjy** [comment]
Still document page of [`_get_property_list`](https://docs.godotengine.org/en/stable/classes/class_object.html#method-descriptions) is not documenting about this behavior nor easy to search that one have to reload project or close scene to make an effect.

It's only written in `@tool`'s [try it out](https://docs.godotengine.org/en/stable/tutorials/plugins/running_code_in_the_editor.html#try-tool-out) section, which still says *"reload the scene"* if it doesn't work.

However this doesn't cover the case where script is resource - which requires project refresh instead as we can't close resources.

Either such warning should be documented in `_get_property_list`, or bug should be fixed I think, as this still happens (Godot 4.3 beta 1 public) and took 2 hours for me either. Consider how much more hours we could save in utilitarianism perspective!
